X_VCPKG_APPLOCAL_DEPS_INSTALL=ON fails if object libraries are installed

Consider a project that installs object libraries. For example, Qt does this for its examples asset library.

```
add_library(ExampleIconsPrivate_resources_1 OBJECT  ...)
...
install(TARGETS ExampleIconsPrivate_resources_1 ...)
```

Configure such a project with `X_VCPKG_APPLOCAL_DEPS_INSTALL=ON`. This fails with:

```
CMake Error:
  Error evaluating generator expression:

    $<TARGET_FILE_NAME:ExampleIconsPrivate_resources_1>

  Target "ExampleIconsPrivate_resources_1" is not an executable or library.
```

This is triggered in `x_vcpkg_install_local_dependencies`:
```
        foreach(target IN LISTS arg_TARGETS)
            get_target_property(target_type "${target}" TYPE)
            if(NOT target_type STREQUAL "INTERFACE_LIBRARY")
                install(CODE "message(\"-- Installing app dependencies for ${target}...\")
                    execute_process(COMMAND \"${Z_VCPKG_POWERSHELL_PATH}\" -noprofile -executionpolicy Bypass -file \"${Z_VCPKG_TOOLCHAIN_DIR}/msbuild/applocal.ps1\"
                        -targetBinary \"${arg_DESTINATION}/$<TARGET_FILE_NAME:${target}>\"
                        -installedDir \"${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}$<$<CONFIG:Debug>:/debug>/bin\"
                        -OutVariable out)"
                    ${component_param}
                )
            endif()
        endforeach()
```

The code above excludes interface libraries, but it must exclude object libraries too.

--- a/scripts/buildsystems/vcpkg.cmake
+++ b/scripts/buildsystems/vcpkg.cmake
@@ -699,9 +699,10 @@ function(x_vcpkg_install_local_dependencies)
             set(component_param COMPONENT "${arg_COMPONENT}")
         endif()
 
+        set(allowed_target_types MODULE_LIBRARY SHARED_LIBRARY EXECUTABLE)
         foreach(target IN LISTS arg_TARGETS)
             get_target_property(target_type "${target}" TYPE)
-            if(NOT target_type STREQUAL "INTERFACE_LIBRARY")
+            if(target_type IN_LIST allowed_target_types)
                 install(CODE "message(\"-- Installing app dependencies for ${target}...\")
                     execute_process(COMMAND \"${Z_VCPKG_POWERSHELL_PATH}\" -noprofile -executionpolicy Bypass -file \"${Z_VCPKG_TOOLCHAIN_DIR}/msbuild/applocal.ps1\"
                         -targetBinary \"${arg_DESTINATION}/$<TARGET_FILE_NAME:${target}>\"

